src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -127,6 +128,7 @@ export class TerminalService implements ITerminalService {
 			terminalInstance.setVisible(i === terminalIndex);
 		});
 		this._onActiveInstanceChanged.fire();
+		this.getActiveInstance().focus(true);
```

Comment:
@Tyriar Ok, I moved it into the block I had previously in mind, the first two isolation conditions are already met there. The third one was a little trickier because the last focus state had to be remembered, therefor https://github.com/Microsoft/vscode/pull/15958/commits/c46e28401f1ec094f82d47ab3f2330359766bdd9 to cover this case.